test/e2e/specs/editor/blocks/links.spec.js

```diff
@@ -8,6 +8,537 @@ test.describe( 'Links', () => {
+			page.locator(
+				'.components-popover__content .block-editor-link-control'
+			)
+		).toBeVisible();
+
+		// Trigger the autocomplete suggestion list and select the first suggestion.
+		await page.keyboard.type( titleText );
+		await expect(
+			page.getByRole( 'option', {
+				name: titleText + ' localhost:8889/?p=' + postId + ' post',
+			} )
+		).toBeVisible();
+		await page.keyboard.press( 'ArrowDown' );
+
+		// Expect the escape key to dismiss the popover when the autocomplete suggestion list is open.
+		await page.keyboard.press( 'Escape' );
+		await expect(
+			page.locator(
+				'.components-popover__content .block-editor-link-control'
+			)
+		).not.toBeVisible();
+
+		// Confirm that selection is returned to where it was before launching
+		// the link editor, with "Gutenberg" as an uncollapsed selection.
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.type( '.' );
+
+		await expect.poll( editor.getBlocks ).toMatchObject( [
+			{
+				name: 'core/paragraph',
+				attributes: {
+					content: 'This is Gutenberg.',
+				},
+			},
+		] );
+
+		// Press Cmd+K to insert a link.
+		await pageUtils.pressKeys( 'primary+K' );
+
+		await expect(
+			page.locator(
+				'.components-popover__content .block-editor-link-control'
+			)
+		).toBeVisible();
+
+		// Expect the escape key to dismiss the popover normally.
+		await page.keyboard.press( 'Escape' );
+		await expect(
+			page.locator(
+				'.components-popover__content .block-editor-link-control'
+			)
+		).not.toBeVisible();
+
+		// Press Cmd+K to insert a link.
+		await pageUtils.pressKeys( 'primary+K' );
+
+		await expect(
+			page.locator(
+				'.components-popover__content .block-editor-link-control'
+			)
+		).toBeVisible();
+
+		// Tab to the "Open in new tab" toggle.
+		await page.keyboard.press( 'Tab' );
+		await page.keyboard.press( 'Tab' );
+
+		// Expect the escape key to dismiss the popover normally.
+		await pageUtils.pressKeys( 'Escape' ); //page.keyboard.press( 'Escape' );
+		await expect(
+			page.locator(
+				'.components-popover__content .block-editor-link-control'
+			)
+		).not.toBeVisible();
+	} );
+
+	test( `can be modified using the keyboard once a link has been set`, async ( {
+		page,
+		editor,
+		pageUtils,
+	} ) => {
+		const URL = 'https://wordpress.org/gutenberg';
+
+		// Create a block with some text and format it as a link.
+		await editor.insertBlock( {
+			name: 'core/paragraph',
+		} );
+		await page.keyboard.type( 'This is Gutenberg' );
+		await pageUtils.pressKeys( 'shiftAlt+ArrowLeft' );
+		await pageUtils.pressKeys( 'primary+K' );
+		await page.keyboard.type( URL );
+		await pageUtils.pressKeys( 'Enter' );
+
+		// Deselect the link text by moving the caret to the end of the line
+		// and the link popover should not be displayed.
+		await pageUtils.pressKeys( 'End' );
+		await expect(
+			page.locator(
+				'.components-popover__content .block-editor-link-control'
+			)
```

Comment:
Let's move this to an abstraction in order that we can refactor one https://github.com/WordPress/gutenberg/pull/54063 is merged and we have better selectors availabke.

Reply:
✅